server/catgenome/src/main/java/com/epam/catgenome/manager/FeatureIndexManager.java

```diff
@@ -320,10 +321,27 @@ public int getTotalPagesCount(VcfFilterForm filterForm, long projectId) throws I
      * @throws IOException
      */
     public IndexSearchResult<VcfIndexEntry> filterVariations(VcfFilterForm filterForm) throws IOException {
-        List<VcfFile> files = vcfFileManager.loadVcfFiles(filterForm.getVcfFileIds());
+        List<VcfFile> files = vcfFileManager.loadVcfFiles(filterForm.getVcfFileIds())
+                .stream().filter(file -> filterVcfFile(file, filterForm)).collect(Collectors.toList());
         return getVcfSearchResult(filterForm, files);
     }
 
+    /**
+     * Filter vcf files according to the additional filter, specified by VCF file names
+     * @param file VCF file instance in a database
+     * @param filterForm {@code VcfFilterForm}, setting filter options
+     * @return a {@link List} of {@link VcfFile}s, representing VCF files that satisfy the additional filter
+     */
+    private boolean filterVcfFile(VcfFile file, VcfFilterForm filterForm) {
+        String additionalFilterName = FeatureIndexDao.FeatureIndexFields.SOURCE_FILE.getFieldName();
+        if(filterForm.getAdditionalFilters() != null
+                && filterForm.getAdditionalFilters().containsKey(additionalFilterName)) {
+            List<String> fileNames = (List<String>) filterForm.getAdditionalFilters().get(additionalFilterName);
```

Comment:
Why do we check filterForm.getAdditionalFilters() instead of filterForm.getVcfFilesName() ?
Also call to contains() method on a list is a bad practice.